src/legilogic_ethereum/ethereum_user.ml

```diff
@@ -598,26 +607,70 @@ let transfer_gas_used = TokenAmount.of_int 21000
 let transfer_tokens ~recipient value =
   PreTransaction.{operation=(Operation.TransferTokens recipient); value; gas_limit=transfer_gas_used}
 
-let make_pre_transaction ~sender operation ?gas_limit value : PreTransaction.t Lwt_exn.t =
+let make_pre_transaction ~sender operation ?gas_limit ~value : PreTransaction.t Lwt_exn.t =
```

Comment:
If value may have a default value if that makes sense (probably not), but is not an option.